cranelift/filetests/filetests/isa/aarch64/call.clif

```diff
@@ -905,30 +903,32 @@ block0(v0: i64):
 ; VCode:
 ;   stp fp, lr, [sp, #-16]!
 ;   mov fp, sp
-;   str x24, [sp, #-16]!
+;   str x23, [sp, #-16]!
 ; block0:
-;   mov x24, x8
-;   load_ext_name x4, TestCase(%g)+0
-;   blr x4
-;   mov x8, x24
-;   ldr x24, [sp], #16
+;   mov x23, x8
+;   load_ext_name x2, TestCase(%g)+0
+;   mov x8, x23
+;   blr x2
+;   mov x8, x23
```

Comment:
This was an interesting one! The call will clobber the value in `x8`, but we require that value to be live to the end of the function so that we can return it, so RA2 rewrites the input constraint on the call to an `any` constraint and moves into `x8` right before the call. This ensures that we have a value (`x23`) to restore `x8` after the call, but does yield a slightly funny result. 

The move from `x23` into `x8` on line 910 could probably be eliminated by the redundant move eliminator, but we could also look into resolving the need for the value duplication on the in-progress overlap support PR...

Reply:
That does fully explain this, doesn't it? Yeah, I guess I'm not too worried about an extra move that is:
- before a call
- from a function with an sret parameter
- to another function with an sret parameter

Especially since this sounds like exactly the sort of situation that the RA2 overlapping live ranges work should fix.

Reply:
(Just for posterity, here's the location in RA2 where this rewrite happens)
https://github.com/bytecodealliance/regalloc2/blob/244ede83638762701831453a43e6011a7dbe0e94/src/ion/liveranges.rs#L557-L616